src/Agent.Sdk/Util/WindowsProcessUtil.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+
+namespace Microsoft.VisualStudio.Services.Agent.Util
+{
+    internal record ProcessInfo(int ProcessId, string ProcessName);
+
+    public static class WindowsProcessUtil
+    {
+        internal static ProcessInfo GetParentProcessInformation(int processId)
+        {
+            using var query = new ManagementObjectSearcher($"SELECT ParentProcessId FROM Win32_Process WHERE ProcessId={processId}");
+
+            using ManagementObjectCollection queryResult = query.Get();
+
+            using ManagementObject foundProcess = queryResult.OfType<ManagementObject>().FirstOrDefault();
```

Comment:
Consider using non-blocking query to prevent deadlocks:
https://learn.microsoft.com/en-us/mem/configmgr/develop/core/clients/programming/how-to-perform-an-asynchronous-query-by-using-system.management